src/common/Timer.cc

```diff
@@ -126,8 +126,11 @@ void SafeTimer::add_event_after(double seconds, Context *callback)
 void SafeTimer::add_event_at(utime_t when, Context *callback)
 {
   assert(lock.is_locked());
-  ldout(cct,10) << "add_event_at " << when << " -> " << callback << dendl;
-
+  ldout(cct,10) << __func__ << " " << when << " -> " << callback << dendl;
+  if (!thread) {
```

Comment:
there's a window when the lock isn't held during shutdown but the thread is joined, and not set to NULL yet - checking for stopping here instead should avoid that